dts/bindings/base/pm.yaml

```diff
@@ -30,3 +30,8 @@ properties:
     description: |
       Automatically configure the device for runtime power management after the
       init function runs.
+
+  zephyr,disabling-power-states:
```

Comment:
I don't understand how the device drivers could use these details. When a device driver is activated, it locks all power states, resulting in clock loss (including those with loss of registers). When a device driver is deactivated, it unlocks all power states (including these with loss of registers). When the device driver gets activated again, it has no information on which power states were entered meantime, so it must assume the worst - the registers' values were lost and must be restored.

Reply:
>  When the device driver gets activated again, it has no information on which power states were entered meantime, so it must assume the worst - the registers' values were lost and must be restored.

If registers where not lost we should not take time to restore them at resumption. Restoring registers take time it should not be done by default.
This is why we have various power states with min-residency-us and exit-latency-us for each state. Depending on the deepness of the power state, more actions have to be taken at resumption. Based on these information, PM subsystem arbitrate the next sleep state to enter depending on the idle time available.

Reply:
I agree that we should avoid time-consuming restoring if possible. But I don't see from where a device driver could take the information if restoring is needed. Do you have a proposal on how to modify the framework to provide this information to the drivers?

Reply:
See https://github.com/zephyrproject-rtos/zephyr/pull/69249#issuecomment-1971368367

Reply:
Fair enough. The notification function misses the substate information, but I think it can be fixed.
With this context, I agree more detailed properties in the device tree nodes would be useful. Right now I'm mostly interested in the property describing power states resulting in the loss of clock (not necessarily register values).

Reply:
That is something I will fix. The problem that this pr address is not achievable with notifications because the notification is sent after the policy has chosen which state the SoC will use. This pr allows a device set constraints in these states to avoid them to be used.